src/management/list.ts

```diff
@@ -35,9 +36,71 @@ export async function listFirebaseProjects(
   }
 }
 
+/**
+ * Send recurring API requests to list all Firebase apps of a Firebase project
+ * @return a promise that resolves to the list of all Firebase apps.
+ */
+export async function listFirebaseApps(
+  projectId: string,
+  options: { platform?: AppPlatform; pageSize?: number } = {}
+): Promise<AppMetadata[]> {
+  const apps: AppMetadata[] = [];
+  const { platform, pageSize } = options;
+  try {
+    let nextPageToken = "";
+    do {
+      const response = await getPageApiRequest(
+        getListAppsResourceString(projectId, platform),
+        pageSize || APP_LIST_PAGE_SIZE,
+        nextPageToken
+      );
+      if (response.body.apps) {
+        const appsOnPage = response.body.apps.map(
+          // app.platform does not exist if we use the endpoint for a specific platform
+          (app: any) => (app.platform ? app : { ...app, platform })
+        );
+        apps.push(...appsOnPage);
+      }
+      nextPageToken = response.body.nextPageToken;
+    } while (nextPageToken);
+
+    return apps;
+  } catch (err) {
+    logger.debug(err.message);
+    throw new FirebaseError(
+      `Failed to list Firebase ${platform ? platform + " " : ""}` +
+        "apps. See firebase-debug.log for more info.",
+      {
+        exit: 2,
+        original: err,
+      }
+    );
+  }
+}
+
+function getListAppsResourceString(projectId: string, platform?: AppPlatform): string {
+  let resourceSuffix;
+  switch (platform) {
+    case AppPlatform.IOS:
+      resourceSuffix = "/iosApps";
+      break;
+    case AppPlatform.ANDROID:
+      resourceSuffix = "/androidApps";
+      break;
+    case AppPlatform.WEB:
+      resourceSuffix = "/webApps";
+      break;
+    default:
+      resourceSuffix = ":searchApps";
+      break;
+  }
+
+  return `/v1beta1/projects/${projectId}${resourceSuffix}`;
+}
+
 async function getPageApiRequest(
   resource: string,
-  pageSize: number = PROJECT_LIST_PAGE_SIZE,
+  pageSize: number,
```

Comment:
I think having the default here is better than having to default it (with `||`) above. Any particular reason why this was removed?

Reply:
Since we are having `APP_LIST_PAGE_SIZE` and `PROJECT_LIST_PAGE_SIZE`, it doesn't make sense anymore to default to the later.
(Since we agreed to have `projects.ts` and `apps.ts`, I will probably introduce the default value again after splitting `listFirebaseProject()` and `listFirebaseApps()` from `list.ts